src/log/iobuf.rs

```diff
@@ -426,13 +441,32 @@ impl IoBufs {
         );
 
         let res_len = offset(header) as usize;
+
+        #[cfg(feature = "o_direct_writer")]
+        let pad_len = {
+            let remainder = res_len % 512;
+            if remainder == 0 { 0 } else { 512 - remainder }
+        };
```

Comment:
here we have 2 choices: add a bunch of padding in-between log entries, or do extra buffering so we can keep things extra squished. Adding some padding is faster, and still correct since our log iterator will just skip over zeroed out entries anyway.

Reply:
(must do everything on the 512 (or some other) byte boundaries with O_DIRECT!)